configs/provisioner.go

```diff
@@ -50,6 +50,11 @@ func decodeProvisionerBlock(block *hcl.Block) (*Provisioner, hcl.Diagnostics) {
 		}
 	}
 
+	// destroy provisioners can only refer to self
```

Comment:
Tiny nitpick/question:  since we're only giving warnings right now should this comment say `destroy provisioners _should_ only refer to self`? 
I am completely ok with disagreement :)

Reply:
Even though that was only there because I backported this from it being an error, I think it might be better as "can". In this case, they "_can_ only refer to self" in order to avoid the deprecation warnings?

Reply:
Nice clarification, makes sense to me (and I wasn't going to fight you on it anyway)!